### PR TITLE
build: fix eslint

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

* Makes ESLint run on `.ts`/`.tsx` files
* Corrects newly detected linting errors
* Disables weird ESLint rules inherited from Airbnb

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by beheh** [comment]
I'll look at this this weekend, sorry for the delay.

File: src/instruments/src/EFB/Components/Form/Input/Input.tsx [hunk 1]
```diff
@@ -16,9 +16,11 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, {useEffect, useState} from "react";
+/* eslint-disable react/require-default-props */
```
Comment (beheh) [inline src/instruments/src/EFB/Components/Form/Input/Input.tsx:19]: Let's just disable `react/require-default-props` in Typescript files (via overrides).

File: src/instruments/src/EFB/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "ignorePatterns": [
+    "web/**"
+  ]
+}
```
Comment (beheh) [inline src/instruments/src/EFB/.eslintrc:5]: Missing trailing new line.

File: src/instruments/src/EFB/Dispatch/Dispatch.tsx [hunk 3]
```diff
@@ -69,76 +69,77 @@ type DispatchState = {
...
-       'Overview',
-       'Loadsheet',
-       'Fuel',
-       'Payload',
-    ];
-
+    // eslint-disable-next-line react/state-in-constructor
     state: DispatchState = {
         activeIndex: 0,
     };
```
Comment (beheh) [inline src/instruments/src/EFB/Dispatch/Dispatch.tsx:87]: Why bypass this? This should just be moved to the constructor.

File: src/instruments/src/EFB/Dispatch/Pages/OverviewPage.tsx [hunk 4]
```diff
@@ -16,8 +16,12 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, {useState, useEffect} from 'react';
-import { IconAlignRight, IconBox, IconPlane, IconSwitchHorizontal, IconUsers, IconBolt } from '@tabler/icons';
+/* eslint-disable */
```
Comment (beheh) [inline src/instruments/src/EFB/Dispatch/Pages/OverviewPage.tsx:19]: Why is this file disabled?

File: src/instruments/src/EFB/Efb.tsx [hunk 5]
```diff
@@ -72,6 +72,7 @@ class Efb extends React.Component<EfbProps, EfbState> {
         this.fetchSimbriefData = this.fetchSimbriefData.bind(this);
     }
 
+    // eslint-disable-next-line react/state-in-constructor
```
Comment (beheh) [inline src/instruments/src/EFB/Efb.tsx:75]: Once again, this should just be moved to the constructor.

File: src/instruments/src/EFB/Efb.tsx [hunk 6]
```diff
@@ -114,117 +116,118 @@ class Efb extends React.Component<EfbProps, EfbState> {
...
 
     async fetchSimbriefData() {
         if (!this.state.simbriefUsername) {
             return;
         }
 
-        console.log("Fetching simbriefData");
+        console.log('Fetching simbriefData');
+        // eslint-disable-next-line react/no-access-state-in-setstate
         const simbriefData = await getSimbriefData(this.state.simbriefUsername);
```
Comment (beheh) [inline src/instruments/src/EFB/Efb.tsx:144]: I'm confused why this is necessary. Is this a false positive?

File: src/instruments/src/EFB/SimbriefApi/simbriefInterface.ts [hunk 7]
```diff
@@ -16,6 +16,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+/* eslint-disable camelcase */
```
Comment (beheh) [inline src/instruments/src/EFB/SimbriefApi/simbriefInterface.ts:19]: I'd be more inclined to just rename this and drop the hungarian notation, but I see why this PR itself should maybe not do that and rather just bypass the rule.

File: src/instruments/src/EFB/StatusBar/StatusBar.tsx [hunk 8]
```diff
@@ -33,9 +32,11 @@ type TimeState = {
 }
 
 export default class StatusBar extends React.Component<TimeProps, any> {
+    // eslint-disable-next-line react/state-in-constructor
```
Comment (beheh) [inline src/instruments/src/EFB/StatusBar/StatusBar.tsx:35]: Move to constructor.

File: src/instruments/src/EFB/SimbriefApi/simbriefParser.ts [hunk 9]
```diff
@@ -16,71 +16,72 @@
...
 
-import { ISimbriefData } from "./simbriefInterface";
+import { ISimbriefData } from './simbriefInterface';
 
-const simbriefApiUrl: URL = new URL("https://www.simbrief.com/api/xml.fetcher.php");
+const simbriefApiUrl: URL = new URL('https://www.simbrief.com/api/xml.fetcher.php');
 const simbriefApiParams = simbriefApiUrl.searchParams;
 
+// eslint-disable-next-line no-undef
 const getRequestData: RequestInit = {
```
Comment (beheh) [inline src/instruments/src/EFB/SimbriefApi/simbriefParser.ts:25]: Why is this triggered? Because of RequestInit?

**Review by beheh — CHANGES_REQUESTED** [review]
I've added a few comments. The most obvious one is disabling the rules about state in the constructor, I think this PR should just move those into the or a new constructor.

From the rules side we're mostly fine, however:
- Let's disable `react/no-unused-prop-types` and `react/no-unused-state`. We can be more strict about this later, but for now while we're implementing some system piece by this I don't want people removing vars we might need later.
- Let's disable `react/require-default-props` via `overrides` in `.tsx` files. Default props are not really a thing in TS, as you will be forced to either define them or handle the undefined in your component.

File: src/instruments/src/EFB/SimbriefApi/simbriefInterface.ts (hunk 7 above)
Comment (Benjozork) [reply src/instruments/src/EFB/SimbriefApi/simbriefInterface.ts:19]: this is parsing data from an API

File: src/instruments/src/EFB/StatusBar/StatusBar.tsx (hunk 8 above)
Comment (Benjozork) [reply src/instruments/src/EFB/StatusBar/StatusBar.tsx:35]: Resolved.

File: src/instruments/src/EFB/Efb.tsx (hunk 5 above)
Comment (Benjozork) [reply src/instruments/src/EFB/Efb.tsx]: Resolved.

File: src/instruments/src/EFB/SimbriefApi/simbriefParser.ts (hunk 9 above)
Comment (Benjozork) [reply src/instruments/src/EFB/SimbriefApi/simbriefParser.ts:25]: adding `browser` to `env` should resolve this, but doesn't. this should get merged first

File: src/instruments/src/EFB/Dispatch/Dispatch.tsx (hunk 3 above)
Comment (Benjozork) [reply src/instruments/src/EFB/Dispatch/Dispatch.tsx]: Done

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 10]
```diff
@@ -24,40 +24,42 @@ type SettingsState = {
 }
 
 class Settings extends React.Component<SettingsProps, SettingsState> {
+    // eslint-disable-next-line react/state-in-constructor
```
Comment (beheh) [inline src/instruments/src/EFB/Settings/Settings.tsx:27]: Missed this one.

File: src/instruments/src/EFB/Components/Form/Input/Input.tsx (hunk 1 above)
Comment (Benjozork) [reply src/instruments/src/EFB/Components/Form/Input/Input.tsx]: Done

File: src/instruments/src/EFB/.eslintrc (hunk 2 above)
Comment (Benjozork) [reply src/instruments/src/EFB/.eslintrc:5]: Done

File: src/instruments/src/EFB/Settings/Settings.tsx (hunk 10 above)
Comment (Benjozork) [reply src/instruments/src/EFB/Settings/Settings.tsx]: Done

**Review by beheh — APPROVED** [review]

**Pull request merged**